src/test/java/org/jfrog/hudson/pipeline/integration/PipelineTestBase.java

```diff
@@ -76,29 +77,37 @@
 
     @BeforeClass
     public static void setUp() {
+        currentTime = System.currentTimeMillis();
         verifyEnvironment();
         createSlave();
         setJarsLibEnv();
         createClients();
         cleanUpArtifactory(artifactoryClient);
         createPipelineSubstitution();
+        // Create repositories
+        Arrays.stream(TestRepository.values()).forEach(PipelineTestBase::createRepo);
     }
 
     @Before
     public void beforeTest() throws IOException {
         log.info("Running test: " + pipelineType + " / " + testName.getMethodName());
-        // Create repositories
-        Arrays.stream(TestRepository.values()).forEach(this::createRepo);
         FileUtils.cleanDirectory(testTemporaryFolder.getRoot().getAbsoluteFile());
     }
 
     @After
-    public void deleteRepos() {
-        Arrays.stream(TestRepository.values()).forEach(repoName -> artifactoryClient.repository(getRepoKey(repoName)).delete());
+    public void CleanRepos() {
+        // Reemove the content of all local repositories
+        Arrays.stream(TestRepository.values()).filter(repoName -> repoName.getRepoName().contains("local"))
```

Comment:
We're deleting here the content of the local repositories. The code determines that a repository is local, if its name includes "local". This code may fail in the future, if a local repo is added with a name that does not follow this guideline.
I therefore suggest using a different approach. We can maintain a list of all the local repos. The list of all repos will include this list.
The same goes for the code in the *tearDown()* method.

Reply:
Since we need to filter the repo list according to types (in clean up we need the local repos, in tear down we need the virtual etc.), I think adding a type field for each repo is more easy to maintain.
Please See my fix.